api/LsLink.py

```diff
@@ -0,0 +1,217 @@
+
+    def __init__(self, version, data):
+        """
+        Handle the message by parsing it and storing the data in memory.
+
+        :param version: Schema version of the message.
+        :param data: Data to parse.
+        """
+
+        super(LsLink, self).__init__()
+        self.spec_version = version
+
+        if version >= float(1.3):
+
+            self.headerNames = ["action", "seq", "hash", "base_attr_hash", "router_hash", "router_ip", "peer_hash", "peer_ip",
+                "peer_asn", "timestamp", "igp_router_id", "router_id", "routing_id", "ls_id",
+                "ospf_area_id", "isis_area_id", "protocol", "as_path", "local_pref", "med", "nexthop",
+                "mt_id", "local_link_id", "remote_link_id", "intf_ip", "nei_ip", "igp_metric",
+                "admin_group", "max_link_bw", "max_resv_bw", "unresv_bw", "te_default_metric",
+                "link_protection", "mpls_proto_mask", "srlg", "link_name", "remote_node_hash",
+                "local_node_hash", "remote_igp_router_id", "remote_router_id", "local_node_asn",
+                "remote_node_asn", "peer_node_sid", "isPrePolicy", "isAdjRibIn"]
+
+        elif version >= float(1.2):
+
+            self.headerNames = ["action", "seq", "hash", "base_attr_hash", "router_hash", "router_ip", "peer_hash", "peer_ip",
+                "peer_asn", "timestamp", "igp_router_id", "router_id", "routing_id", "ls_id",
+                "ospf_area_id", "isis_area_id", "protocol", "as_path", "local_pref", "med", "nexthop",
+                "mt_id", "local_link_id", "remote_link_id", "intf_ip", "nei_ip", "igp_metric",
+                "admin_group", "max_link_bw", "max_resv_bw", "unresv_bw", "te_default_metric",
+                "link_protection", "mpls_proto_mask", "srlg", "link_name", "remote_node_hash",
+                "local_node_hash", "remote_igp_router_id", "remote_router_id", "local_node_asn",
+                "remote_node_asn", "peer_node_sid"]
+
+        else:
+
+            self.headerNames = ["action", "seq", "hash", "base_attr_hash", "router_hash", "router_ip", "peer_hash", "peer_ip",
+                "peer_asn", "timestamp", "igp_router_id", "router_id", "routing_id", "ls_id",
+                "ospf_area_id", "isis_area_id", "protocol", "as_path", "local_pref", "med", "nexthop",
+                "mt_id", "local_link_id", "remote_link_id", "intf_ip", "nei_ip", "igp_metric",
+                "admin_group", "max_link_bw", "max_resv_bw", "unresv_bw", "te_default_metric",
+                "link_protection", "mpls_proto_mask", "srlg", "link_name", "remote_node_hash",
+                "local_node_hash"]
+
+        self.parse(version, data)
+
+    def getProcessors(self):
+        """
+        Processors used for each field.
+        Order matters and must match the same order as defined in headerNames
+
+        :return: array of cell processors.
+        """
+
+        processors = None
+
+        if self.spec_version >= float(1.3):
+
+            processors = [
```

Comment:
Same thing for the processors.   Let's change it so that we only add the extra processors instead of duplicating it.